k4FWCore/include/k4FWCore/Consumer.h

```diff
@@ -98,7 +98,7 @@ namespace k4FWCore {
        * @param i  The index of the input
        * @return   A range of the input locations
        */
-      const auto inputLocations(int i) const {
+      auto inputLocations(unsigned i) const {
```

Comment:
how about using size_t since this is later compared with sizeof... (gives size_t) and used as index?
```suggestion
      auto inputLocations(size_t i) const {
```